hackathon_site/event/api_views.py

```diff
@@ -66,3 +81,57 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_201_CREATED,)
+
+
+class JoinTeamView(generics.GenericAPIView, mixins.RetrieveModelMixin):
+    permission_classes = [UserHasProfile]
+    serializer_class = TeamSerializer
+
+    def clean(self):
+        if not is_registration_open():
+            raise forms.ValidationError(
+                _("You cannot change teams after registration has closed."),
+                code="registration_closed",
+            )
+
+        return super().clean()
+
+    @transaction.atomic
+    def post(self, request, team_code, *args, **kwargs):
+
+        profile = request.user.profile
+        current_team = profile.team
+
+        try:
+            team = Team.objects.get(team_code=team_code)
+        except:
+            raise ValidationError(
+                {"detail": "Team does not exist!"}, code=status.HTTP_400_BAD_REQUEST
+            )
+
+        if not team.profiles.exists():
+            raise ValidationError(
+                {"detail": "Cannot join a fully empty team!"},
+                code=status.HTTP_400_BAD_REQUEST,
+            )
+        if team.profiles.count() >= Team.MAX_MEMBERS:
+            raise ValidationError({"detail": "Team is full"})
+
+        # Raise 400 if team has active orders
```

Comment:
We don't want to let them leave a team if the team they are on has active orders (which is what the ticket says) - I don't see any reason to not let them join a team with orders, they just won't be able to leave it once they join

Reply:
now, members cannot join another team if current team has active orders, as per expectation/description of JIRA issue